internal/ingress/controller/store/store.go

```diff
@@ -624,10 +624,10 @@ func New(
 	return store
 }
 
-// isCatchAllIngress returns whether or not an ingress produces a
+// hasCatchAllIngress returns whether or not an ingress produces a
 // catch-all server, and so should be ignored when --disable-catch-all is set
-func isCatchAllIngress(spec networkingv1beta1.IngressSpec) bool {
-	return spec.Backend != nil && len(spec.Rules) == 0
+func hasCatchAllIngress(spec networkingv1beta1.IngressSpec) bool {
```

Comment:
I'm not sure if this renaming is successful - ingress does not have an ingress for it to be catch all.

What about `hasCatchAllIngressRule`?